test/acceptor_SUITE.erl

```diff
@@ -744,6 +773,32 @@ tcp_echo(_) ->
 	{'EXIT', _} = begin catch ranch:get_port(Name) end,
 	ok.
 
+tcp_local_echo(_) ->
+	case do_os_supports_local_sockets() of
+		true ->
+			do_tcp_local_echo();
+		false ->
+			{skip, "No local socket support."}
+	end.
+
+do_tcp_local_echo() ->
+	doc("Ensure that listening on a local socket works with TCP transport."),
+	Name = name(),
+	SockFile = do_tempname(),
+	{ok, _} = ranch:start_listener(Name,
+		ranch_tcp, #{socket_opts => [{ip, {local, SockFile}}]},
+		echo_protocol, []),
+	0 = ranch:get_port(Name),
+	{ok, Socket} = gen_tcp:connect({local, SockFile}, 0, [binary, {active, false}, {packet, raw}]),
+	ok = gen_tcp:send(Socket, <<"TCP Ranch is working!">>),
+	{ok, <<"TCP Ranch is working!">>} = gen_tcp:recv(Socket, 21, 1000),
+	ok = ranch:stop_listener(Name),
+	{error, closed} = gen_tcp:recv(Socket, 0, 1000),
+	%% Make sure the listener stopped.
+	{'EXIT', _} = begin catch ranch:get_port(Name) end,
+	ok = file:delete(SockFile),
```

Comment:
The delete must be done in an after clause. Put the try before ranch:start_listener, and then conclude with `after ->`. It'll always run including on failure (except when the VM is killed).